sentinel-extension/sentinel-prometheus-exporter/src/main/java/com/alibaba/csp/sentinel/prom/collector/SentinelCollector.java

```diff
@@ -0,0 +1,119 @@
+package com.alibaba.csp.sentinel.prom.collector;
+
+import com.alibaba.csp.sentinel.config.SentinelConfig;
+import com.alibaba.csp.sentinel.log.RecordLog;
+import com.alibaba.csp.sentinel.node.metric.MetricNode;
+import com.alibaba.csp.sentinel.node.metric.MetricSearcher;
+import com.alibaba.csp.sentinel.node.metric.MetricWriter;
+import com.alibaba.csp.sentinel.prom.PromExporterInit;
+import com.alibaba.csp.sentinel.util.PidUtil;
+import io.prometheus.client.Collector;
+import com.alibaba.csp.sentinel.prom.metrics.GaugeMetricFamily;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * The{@link PromExporterInit} the Collector for prometheus exporter.
+ *
+ * @author karl-sy
+ * @date 2023-07-13 21:15
+ * @since 2.0.0
+ */
+public class SentinelCollector extends Collector {
+
+    private final Object lock = new Object();
+    private volatile MetricSearcher searcher;
+
+    private volatile Long lastFetchTime;
+
+    private volatile String appName;
+
+    private volatile String[] types;
+
+    private volatile String identify;
+
+    private volatile int fetchSize;
+
+    private volatile int delayTime;
+
+    @Override
+    public List<MetricFamilySamples> collect() {
+        int ONE_SECOND = 1000;
+        if (searcher == null) {
+            synchronized (lock) {
+                fetchSize = Integer.parseInt(System.getProperty("sentinel.prometheus.size","1024"));
```

Comment:
You may need to illustrate the attributes in the design doc.

Reply:
In Sentinel, we retrieve attributes via `SentinelConfig.getConfig(key)`.

Reply:
in README.md